### PR TITLE
Bump version of Redis: `5.0.6` -> `7.0.5`

### PR DESCRIPTION
**Pull request by craciunoiuc**
This PR updates the Redis version used by Unikraft from `5.0.6` to `7.0.5`.
This is a 3-year fast-forward of Redis.
This also changes the source used by Redis to the official page.

The changes come from some inconsistencies between using an older server and a newer client for Redis.

Currently, with the PR applied, redis builds and runs successfully in both server and client mode.
It is not guaranteed that it works in all cases, as there are no test cases, but usual operations seem to work.

Superseeds: #2 #3

### CODE REVIEW

**Comment by craciunoiuc** [comment]
Most of the code inside the client now depends on on sources from the server (even `server.c`). As such, I had to move everything into the `common` part of the `Makefile`.

File: patches/0008-src-Disable-unused-cpu-affinity-setting.patch [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+diff --git a/src/setcpuaffinity.c b/src/setcpuaffinity.c
+index 77b162103..c9d221adc 100644
+--- a/src/setcpuaffinity.c
++++ b/src/setcpuaffinity.c
+@@ -29,6 +29,8 @@
+ #include <stdlib.h>
+ #include <string.h>
+ #include <ctype.h>
++
++#define __linux__
```
Comment (mogasergiu) [inline patches/0008-src-Disable-unused-cpu-affinity-setting.patch:20]: Would it not be better to simply delete the other `#ifdef`'s? Or maybe just place an `#undef` after the `#endif`? I know the result is the same, but IMHO it does not reflect the same intention that the upstream Redis code has.

For example, from what I can see, you did delete the `#ifdef`'s afterwards anyway.

Comment (craciunoiuc) [reply patches/0008-src-Disable-unused-cpu-affinity-setting.patch:20]: There's actually `ifdef` guards in `config.h` and that is included after the `ifdefs` you saw (check line 49)

Best I can do is `undef` the symbol afterwards, as it's not used anymore

**Review by mogasergiu — APPROVED** [review]
Reviewed-by: Sergiu Moga <sergiu.moga@protonmail.com>

**Review by mogasergiu — APPROVED** [review]
Reviewed-by: Sergiu Moga <sergiu.moga@protonmail.com>

**Comment by craciunoiuc** [comment]
Did you also test it with musl?

I know there were some problems the last time, that's why we didn't merge this

**Review by mogasergiu — CHANGES_REQUESTED** [review]
> Did you also test it with musl?
> 
> I know there were some problems the last time, that's why we didn't merge this

Nope, I did not re-test it with musl (have yet to get used to using it :) ). I mainly looked at the difference between this and the last version, which involves no functional change. But if you are saying there are problems with musl then yeah it should not get merged and I will take back the R-b.

**Comment by StefanJum** [comment]
Covered by #10

**Pull request closed without merging**